src/lib/ngc/compile-source-files.ts

```diff
@@ -42,14 +42,35 @@ export async function compileSourceFiles(
   // https://github.com/angular/angular/issues/24024
   const emitFlags = tsConfigOptions.declaration ? tsConfig.emitFlags : ng.EmitFlags.JS;
 
-  const result = ng.performCompilation({
+  const scriptTarget = tsConfigOptions.target;
+  const cache = entryPoint.cache;
+  const oldProgram = cache.oldPrograms && cache.oldPrograms[scriptTarget];
+
+  const ngProgram = ng.createProgram({
```

Comment:
Are we no longer using `performCompilation` because it may not emit when there's compilation errors, preventing the program from being reused as per the earlier comment? If so, can that comment and some more explanation go here?